### PR TITLE
Point browser field to esm build

### PR DESCRIPTION
**Pull request by Feiyang1**
Rationale

- Make Firebase work nicely with Angular10.
   Angular10 complains loudly if the browser field points to cjs build. And it's hard for developer to change build configuration which is encapsulated by ngcc
   See https://github.com/firebase/firebase-js-sdk/issues/3315

- Address the incompatibility issue between our typings and esm builds
  Our typings are modeled after the cjs module format, using `export = firebase`, so we ask developers to use namespace import to import firebase, like so: `import * as firebase from ‘firebase/app’`, but it’s not compatible with our esm builds which only have a default export. 
  
  If you manually configure the bundler to use our esm build (e.g. using the `module` field or the` esm2017` field), you will get an error. For example, you will get the following error with webpack:
`"export 'initializeApp' (imported as 'firebase') was not found in 'firebase/app'`

  This change allows developers to use any of our builds without resorting to hacks.

### CODE REVIEW

File: packages/rxfire/package.json
Comment (hsubox76) [inline packages/rxfire/package.json:36]: Does rxfire have a top-level dist/ dir? I couldn't find one after a build. It does have a dist/ and a package.json inside each component subdir.

**Comment by hsubox76** [comment]
I think you have to change this line:
https://github.com/firebase/firebase-js-sdk/blob/eeb1dfa4f629dc5cf328e4b4a224369c0670c312/packages/remote-config/rollup.config.js#L44
to point to `pkg.main` or else you make a weird corrupted esm file which is causing the CI build to fail.

File: packages/component/rollup.config.js
Comment (Feiyang1) [reply packages/component/rollup.config.js:42]: Good catch. reverted.

File: packages/rxfire/package.json
Comment (Feiyang1) [reply packages/rxfire/package.json:36]: hmm, we don't have a top level one. We should remove them all.

**Comment by Feiyang1** [comment]
> I think you have to change this line:
> https://github.com/firebase/firebase-js-sdk/blob/eeb1dfa4f629dc5cf328e4b4a224369c0670c312/packages/remote-config/rollup.config.js#L44
> 
> 
> to point to `pkg.main` or else you make a weird corrupted esm file which is causing the CI build to fail.
> There could possibly be a similar issue in some of the other rollup config files.

Good catch. Updated

**Review by hsubox76 — APPROVED** [review]
Approved pending CI tests.

File: packages/rules-unit-testing/src/api/index.ts
Comment (Feiyang1) [inline packages/rules-unit-testing/src/api/index.ts:27]: @samtstern @yuchenshi 
Meant to ask you before merging the PR - How does developer use these two exports? Do they use the typings from them?
They won't get typings after this change because `firestore` and `database` are not namespaces anymore.

Comment (samtstern) [reply packages/rules-unit-testing/src/api/index.ts:27]: I believe they're used to provide access to Firestore / RTDB classes like this:
https://github.com/firebase/firebase-js-sdk/blob/master/packages/rules-unit-testing/test/database.test.ts#L313

If those tests still pass I think we're fine?  To be honest I don't quite understand namespaces.

Comment (Feiyang1) [reply packages/rules-unit-testing/src/api/index.ts:27]: Yeah, that will continue to work. We are good. Thanks for confirming.

**Pull request merged**
